lib/shopping_cart.rb

```diff
@@ -0,0 +1,51 @@
+class ShoppingCart
+  attr_reader :name,
+              :capacity,
+              :products
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.to_i
+    @products = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    {name: @name, capacity: @capacity}
+  end
+
+  def total_number_of_products
+    @products.inject(0) do |sum, product|
+      sum += product.quantity
+    end
+  end
+
+  def is_full?
+    total_number_of_products >= @capacity
```

Comment:
I think, following IP, this should be a > instead of a >=

Reply:
But I do like that you're utilizing the `total_number_of_products` for this method.